test/support/factories/detour_factory.ex

```diff
@@ -38,12 +38,16 @@ defmodule Skate.DetourFactory do
               "directionId" => sequence(:detour_route_pattern_direction, [0, 1])
             }
           },
-          "value" => %{}
+          "value" => %{},
+          "children" => %{},
+          "historyValue" => %{},
+          "status" => "active"
         }
       end
 
       def with_id(%Skate.Detours.Db.Detour{} = detour, id) do
-        %{detour | state: with_id(detour.state, id)}
+        detour_with_id = %{detour | id: id}
+        %{detour_with_id | state: with_id(detour.state, id)}
```

Comment:
nit: I think this can be done in a single statement?

```suggestion
       %{
          detour
          | id: id,
            state: with_id(detour.state, id)
        }
```